classes/Context.php

```diff
@@ -422,49 +422,15 @@ public function getTranslatorFromLocale($locale)
             (new Filesystem())->remove($cache_file);
         }
 
-        $adminContext = defined('_PS_ADMIN_DIR_');
-        $translator->addLoader('xlf', new XliffFileLoader());
-
-        $sqlTranslationLoader = new SqlTranslationLoader();
-        if (null !== $this->shop) {
-            $sqlTranslationLoader->setTheme($this->shop->theme);
-        }
-
-        $translator->addLoader('db', $sqlTranslationLoader);
-        $notName = $adminContext ? '^Shop*' : '^Admin*';
-
-        $finder = Finder::create()
-            ->files()
-            ->name('*.' . $locale . '.xlf')
-            ->notName($notName)
-            ->in($this->getTranslationResourcesDirectories());
+        // only way to reset the catalogue
+        $translator->addLoader('Array', new ArrayLoader());
+        $translator->addResource('Array', [], $locale);
```

Comment:
This could be better managed inside the translator itself:

```php
public function resetCatalogue($locale)
{
    unset($this->catalogues[$locale]);
}
```

Just add this in the Translator trait and in the PrestaShopTranslatorInterface